File: src/main/scala/com/vc/medium/L311.scala
package com.vc.medium

object L311 {
  def multiply(A: Array[Array[Int]], B: Array[Array[Int]]): Array[Array[Int]] = {
    val nA = A.length
    val mA = A(0).length
    val mB = B(0).length
    val C = Array.ofDim[Int](nA, mB)
    (0 until nA).foreach(i => {
      (0 until mA).foreach(j => {
        if(A(i)(j) != 0) {
          (0 until mB).foreach(k => {
            if(B(j)(k) != 0) C(i)(k) += A(i)(j) * B(j)(k)
          })
        }
      })
    })
    C
  }
}

File: src/main/java/com/vc/medium/L311.java
package com.vc.medium;

class L311 {
    public int[][] multiply(int[][] A, int[][] B) {
        if(A.length == 0 || B.length == 0) return new int[A.length][B.length];

        int[][] res = new int[A.length][B[0].length];
        for(int row = 0; row < A.length; row++) {
            for(int col = 0; col < B[0].length; col++) {
                for(int common = 0; common < B.length; common++) {
                    res[row][col] += A[row][common] * B[common][col];
                }
            }
        }
        return res;
    }
}